paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -245,7 +246,9 @@ def wait_and_terminate(self, slave, drain_timeout, dry_run, region=None):
                         )
                         continue
                     # Check if no tasks are running or we have reached the maintenance window
-                    if is_safe_to_kill(slave.hostname) or dry_run:
+                    if is_safe_to_kill(slave.hostname) or not should_drain:
```

Comment:
I think we want to avoid the `is_safe_to_kill` call completely if `should_drain` is `False`. That way we won't make any calls to the mesos maintenance endpoints at all.